inspirehep/dojson/common/base.py

```diff
@@ -345,3 +345,21 @@ def urls(self, key, value):
             for u in val.get("u"):
                 urls.append({"value": u, "description": description})
     return urls
+
+
+@hepnames.over('field_categories', '^65017')
+@conferences.over('field_categories', '^65017')
+@institutions.over('field_categories', '^65017')
+@jobs.over('field_categories', '^65017')
+@experiments.over('field_categories', '^65017')
+def field_categories(self, key, value):
+    """Field categories."""
+    value = utils.force_list(value)
+    field_categories = self.get('field_categories', [])
+    for element in value:
+        if 'a' in element:
+            el = utils.force_list(element.get('a'))
+            for val in el:
+                corrected_val = classify_field(val)
+                field_categories.append({"value": corrected_val, "_value": val})
```

Comment:
Just wondering, maybe here we could add `if corrected_val:` condition before appending? And `if not corrected_val`, then we log it somewhere (not implemented yet) that record's XML has incorrect structure.